### ISSUE TITLE
Fix the OP wizards

### ISSUE THREAD

**Issue by AshnakAGQ** [ZerdBG3/DnD-Rebalancing#17]
Wizards can currently learn spell scrolls from ANY class including cantrips. They should only be allowed to learn Wizard spells of 1st level and higher.

**Comment by FelipeRenault** [comment]
Not only that, but scrolls should only be usable by classes that have the spell on their class spell list...

> If the spell is on your class's spell list, you can use an action to read the scroll and cast its spell without having to provide any of the spell's components. Otherwise, the scroll is unintelligible.

Although, through my experience with DOS2 modding, I know that it's definitely hard to prevent item usage on this game. I'll see what I can do. For the wizard learning, I think it should be simpler

**Comment by michalnh** [comment]
_"Not only that, but scrolls should only be usable by classes that have the spell on their class spell list... "_
+1

Addition: a wizard, also a cleric, can change memorized/prepared spells every time, not only in camp/rest.

**Comment by LogosWorks** [comment]
Read PHB > Classes > Wizard > Spellbook > Preparing and Casting spells. PHB pg.112

"You can change your list of prepared spells when you finish a long rest. Preparing a new list of wizard spells requires time spent studying your spellbook and memorizing the incantations and gestures you must make to cast the spell: at least 1 minute per spell level for each spell on your list."

Clerics are similar:

Read PHB > Classes > Cleric > Spellcasting > Preparing and Casting spells. PHB pg.36

"You can change your list of prepared spells when you finish a long rest. Preparing a new list of cleric spells requires time spent in prayer and meditation: at least 1 minute per spell level for each spell on your list."

**Comment by FelipeRenault** [comment]
I agree that changing prepared spells on the fly is wrong, but unless the game actually provides a meaningful reason not to spam long rests, this would be just a burden on the player, instead of a mechanic. We'd just be forcing the players to rest everytime they want to swap spells
The only thing this would fix is the spell slot save you get from using Mage Armor or Aid and swapping them right after

**Comment by Merlex-Necromancer** [comment]
As someone who has played D&D Wizards for over 40 years; I agree Wizards should not be able to learn spells not on their spell list. But the Abjurer's _Arcane _Ward__ is of greater concern to me. Seriously 3 hp that doesn't stack with temporary hit points, is a crap ability. Arcane Ward is suppose to be an Abjuration Wizard's premier ability. The correct version is (from PHB): 
**Starting at 2nd level, you can weave magic around yourself for protection. When you cast an abjuration spell of 1st level or higher, you can simultaneously use a strand of the spell’s magic to create a magical ward on yourself that lasts until you finish a long rest. The ward has a hit point maximum equal to twice your wizard level + your Intelligence modifier. Whenever you take damage, the ward takes the damage instead. If this damage reduces the ward to 0 hit points, you take any remaining damage.

While the ward has 0 hit points, it can’t absorb damage, but its magic remains. Whenever you cast an abjuration spell of 1st level or higher, the ward regains a number of hit points equal to twice the level of the spell.

Once you create the ward, you can’t create it again until you finish a long rest.**

It is also suppose to stack with temporary hit point from spells such as _False Life_. Three hit points, why wouldn't I just play an Evoker and use False Life?

**Comment by LogosWorks** [comment]
Not stacking with temp hp is kinda the whole thing that makes it good.
It allows them to combine spells together
Before you complain that BG3-5e is not like 5e you should learn 5e.
warlock2 abjure X is an insane build for grabbing AoA, or using divine sorc for aid.
these are staple 5e builds.

**Comment by FelipeRenault** [comment]
Calm down @LogosWorks , I'm pretty sure Merlex agrees with you on the subject of stacking with Temp HP. You both are not in disagreement in any way.
And please avoid being aggressive, even if they were wrong and you disagreed. I want this space to be a place for people to voice their opinions in any way, even if they're new to DnD and might get some things wrong, which is not the case here

**Comment by FelipeRenault** [comment]
Update on this issue:
- I couldn't prevent scroll usage for spells outside the character's spell list
- I can prevent learning from scrolls, but it's either prevent all or prevent nothing
- I can provide a half-fix for Arcane Ward: I can make it give the appropriate HP at first cast (and maybe the correct value based on spell level), but I couldn't make it have it's own HP. I need to use it as a Temporary HP :x

You guys reckon it's worth to make the changes to Arcane Ward?

**Comment by gamerguyg** [comment]
@FelipeRenault While the arcane ward fix isn't perfect, its better than nothing. I think I'm on character #5 now (my 2nd wizard, Gale has been in my party pretty much 24/7 because their just so strong) and the only time I picked School of Abjuration I reloaded my save cuz its so bad.

**Comment by FelipeRenault** [comment]
Another suggestion is to make scrolls require an Arcana check (DC 10 + Spell Level) when using them to copy the spell to a Wizard's spellbook, as the DMG rules

**Comment by gnufred** [comment]
On the topic of scrolls. I'm confident that this will be addressed in an update. There's just no way Wizard can be balanced if they can learn all spells in the game. It'll come up sooner or later. I mean, Eldritch Blast, Cure Wounds, and so on.
It's still early, early access. I think that's intended for now.

As for cantrips, I doubt it will change. Most players will have no clue of D&D rules and will pickup horrible cantrips, then won't be able to change. So I get the feeling it'll stay like that. But, I hope I'm wrong.

I think we should split this issue into separate issues and put them on hold:

* Learning cantrips from scrolls
* Learning non-wizard spells from scrolls

**Comment by FelipeRenault** [comment]
I'm not so sure this will change.

First, only classes that have the spell on their spell list can cast them from a scroll, and to me, it seems Larian has no intention of implementing that at all, because it goes against some story points (like Speak with Dead and Speak with Animals are spells they want to make available to all characters, regardless of class). Following that logic, I'm assuming they won't implement a check for scroll vs class availability

Secondly, if they want to help new players who picked bad cantrips, it feels unintuitive that only wizards will have that option. It's best to make casters able to change cantrips on level up, if that's the goal.

So, to wrap it up, I agree that those 2 points are actually something I'll have to do eventually, but I'll also need to do one more

- Casting spells from scrolls outside the character's spell list

**Comment by gnufred** [comment]
> Secondly, if they want to help new players who picked bad cantrips, it feels unintuitive that only wizards will have that option. It's best to make casters able to change cantrips on level up, if that's the goal.

I see your point. Makes sense. I agree.
